src/webui/api/torrentscontroller.cpp

```diff
@@ -601,17 +603,20 @@ void TorrentsController::pieceStatesAction()
 void TorrentsController::addAction()
 {
     const QString urls = params()["urls"];
+    const QString cookie = params()["cookie"];
+
     const bool skipChecking = parseBool(params()["skip_checking"]).value_or(false);
     const bool seqDownload = parseBool(params()["sequentialDownload"]).value_or(false);
     const bool firstLastPiece = parseBool(params()["firstLastPiecePrio"]).value_or(false);
     const std::optional<bool> addPaused = parseBool(params()["paused"]);
     const QString savepath = params()["savepath"].trimmed();
     const QString category = params()["category"];
     const QSet<QString> tags = List::toSet(params()["tags"].split(',', QString::SkipEmptyParts));
-    const QString cookie = params()["cookie"];
     const QString torrentName = params()["rename"].trimmed();
-    const int upLimit = params()["upLimit"].toInt();
-    const int dlLimit = params()["dlLimit"].toInt();
+    const int upLimit = parseInt(params()["upLimit"]).value_or(-1);
```

Comment:
BTW, does web api employ rigor parameters checking? For example if `upLimit` is some wrong value `"abc"` should this api return error for it?

Reply:
>BTW, does web api employ rigor parameters checking? 

Now almost none (probably due to the fact that it was originally designed mainly to support the built-in WebUI). But we definitely need to fix this in the new API.